### DISCUSSION TITLE
Preventing stuck particles tutorial page has a bug?

### DISCUSSION THREAD

**Discussion by nikopapa21** [Parcels-code/Parcels#1478] [Q&A]
[](https://docs.oceanparcels.org/en/latest/examples/documentation_unstuck_Agrid.html#2.-Displacement)

specifically after [24]. I get the same error when I implement it to my code.

What is causing this?

**Comment by erikvansebille** [comment]
Thanks for reporting @nikopapa21, I had not seen this error. Fornutately it's easy to solve, just change the definition of the Kernels in cell [24] to
```
kernels = [displace, AdvectionRK4, set_displacement]
```

I'll commit a patch PR asap

**Comment by nikopapa21** [comment]
@erikvansebille Thank you for your reply. Is there a tutorial for preventing stuck particles on a 3D A-grid? I'm having some trouble. Particles are getting negative value depth (above surface) or getting stuck at the ocean floor. I am using these kernels:
```
def KeepInOcean(particle, fieldset, time):
    if particle.state == StatusCode.ErrorThroughSurface:
        particle_ddepth = 1.0
        particle.state = StatusCode.Success

def DeleteParticle(particle, fieldset, time):
    if particle.state == StatusCode.ErrorOutOfBounds: 
        particle.delete()

kernels = [displace, AdvectionRK4_3D, set_displacement, KeepInOcean, DeleteParticle]
```

As I understand, the displacement field is only for the surface layer?

**Reply by erikvansebille** [reply]
No, there is no tutorial for unstuck particles in 3D. By only looking at your code above, I can’t tell you what goes wrong, I’m afraid…
